### PR TITLE
no more tesla stacking

### PR DESCRIPTION
**Pull request by ShadowDeath6**
tesla is way too low effort to set up considering it has easier set up than the singlo with the same or more power generated

you can still put tesla balls in separate pens wide enough for the PA

:cl: ShadowDeath6
rscadd: Tesla energy balls now destabilize and disappear when directly adjacent to one another.
/:cl:

### CODE REVIEW

**Comment by X-TheDark** [comment]
Did you test this?
I'm about 90% sure that this will cause any Tesla to destroy itself, seeing as `/obj/singularity/energy_ball` is both the type of the center tesla and the type of orbiting teslas...

According to the code, if the Tesla has any orbiting balls, it will just break itself, seeing as the orbiting balls are most definitely not the center tesla.

**Comment by ShadowDeath6** [comment]
yeah I know I'm working on it, I just pr'ed it now

**Comment by ShadowDeath6** [comment]
whoops I didn't read mcdonald's comment, that would have saved me like 20 minutes doing trial and error with checks

**Comment by Time-Green** [comment]
Can you have the deleted tesla turn into the orbiting balls of the other tesla? Fun little feature.
Also, are you sure using a for loop with range 40 wont kill the entire server?

**Comment by Alek2ander** [comment]
So, what happens when 2 super-murder-balls collide?
It kills people who are already inside containment or right next to it, so basically nobody who wouldn't have run into the tesla and died anyway. And it makes some spooky sounds.

:-1: unless this becomes something suitably violent.

**Comment by Time-Green** [comment]
Alek, someone in engi hardsuit with insulated is normally tesla proof. So they can still get ashed

**Comment by ShadowDeath6** [comment]
The destabilization will dust any living mob near it, unlike the bolts which won't hurt anybody that's near a grounding rod

@Alek2ander how about making a big emp when they collide

also @Time-Green the SM shard runs a for loop for every living mob

**Comment by ShadowDeath6** [comment]
new commit makes an emp that's really big
it hits psych office apc and puts it at about half energy

**Comment by Iamgoofball** [comment]
lame, at least make it explode

**Comment by ShadowDeath6** [comment]
EMPs are more fun though

**Comment by Iamgoofball** [comment]
do both

**Comment by ShadowDeath6** [comment]
k

**Comment by Cruix** [comment]
Make one tesla chaos dunk the other

**Comment by ShadowDeath6** [comment]
This actually does work so it's r4m unless somebody wants another thing to happen

File: code/modules/power/tesla/energy_ball.dm [hunk 1]
```diff
@@ -51,6 +51,25 @@ var/list/blacklisted_tesla_types = typecacheof(list(/obj/machinery/atmospherics,
 	return ..()
 
 /obj/singularity/energy_ball/process()
+	for(var/obj/singularity/energy_ball/EB in range(1, src))
+		if(EB != src && !orbiting)
+			if(!(EB in src.orbiting_balls))
+				visible_message("<span class='warning'><b>[src]'s electrical field contacts [EB] and destabilizes!</b></span>")
+				for(var/mob/living/L in range(100, src))
```
Comment (Cruix) [inline code/modules/power/tesla/energy_ball.dm]: This range is way too big, loop through the living_mob_list and check if each mob is on the same z-level and check if its distance to the tesla is less than 100 with get_dist(). 100 also seems to be a big large.

File: code/modules/power/tesla/energy_ball.dm [hunk 2]
```diff
@@ -51,6 +51,25 @@ var/list/blacklisted_tesla_types = typecacheof(list(/obj/machinery/atmospherics,
...
+					L << 'sound/effects/supermatter.ogg'
+					L << 'sound/magic/lightningbolt.ogg'
+					L << "<i>Your skin tingles as a wave of energy passes through the air.</i>"
+				for(var/mob/living/L in range(5, src))
```
Comment (Cruix) [inline code/modules/power/tesla/energy_ball.dm]: The explosion is enough, don't dust people too.

Comment (ShadowDeath6) [reply code/modules/power/tesla/energy_ball.dm]: K

**Pull request merged**